### PR TITLE
Reverting dead unallignedCode paths

### PR DESCRIPTION
**Pull request by Anipik**
Reverting unaligned CodePaths were added in https://github.com/dotnet/machinelearning/pull/1218 and https://github.com/dotnet/machinelearning/pull/1274

Related to https://github.com/dotnet/machinelearning/pull/1838

### CODE REVIEW

File: test/Microsoft.ML.TimeSeries.Tests/TimeSeriesEstimatorTests.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public TimeSeriesEstimatorTests(ITestOutputHelper output) : base(output)
         {
         }
 
-        [ConditionalFact(typeof(BaseTestBaseline), nameof(BaseTestBaseline.LessThanNetCore30OrNotNetCore))] // netcore3.0 output differs from Baseline
+        [ConditionalFact(typeof(BaseTestBaseline), nameof(BaseTestBaseline.LessThanNetCore30OrNotNetCore))] // data is not properly alligned for netcoreapp3.0
```
Comment (eerhardt) [inline test/Microsoft.ML.TimeSeries.Tests/TimeSeriesEstimatorTests.cs]: I don't understand the comments on these two tests.

We should change the implementation so it works on any .NET version.

File: test/Microsoft.ML.CpuMath.PerformanceTests/AvxPerformanceTests.cs [hunk 2]
```diff
@@ -11,6 +11,8 @@ namespace Microsoft.ML.CpuMath.PerformanceTests
 {
     public class AvxPerformanceTests : PerformanceTests
     {
+        protected override int allign { get; set; } = 32;
```
Comment (eerhardt) [inline test/Microsoft.ML.CpuMath.PerformanceTests/AvxPerformanceTests.cs]: type-o `align`.

File: src/Microsoft.ML.CpuMath/AvxIntrinsics.cs [hunk 3]
```diff
@@ -173,118 +193,29 @@ public static unsafe void MatMul(ReadOnlySpan<float> mat, ReadOnlySpan<float> sr
...
-                            Vector256<float> mask = Avx.LoadVector256(((float*)(pTrailingAlignmentMask)) + (remainder * 8));
+                        float* pMatTemp = pMatCurrent;
 
-                            float* pMatTemp = pMatCurrent;
-                            Vector256<float> x01 = Avx.And(mask, Avx.LoadVector256(pMatTemp));
-                            Vector256<float> x11 = Avx.And(mask, Avx.LoadVector256(pMatTemp += ccol));
-                            Vector256<float> x21 = Avx.And(mask, Avx.LoadVector256(pMatTemp += ccol));
-                            Vector256<float> x31 = Avx.And(mask, Avx.LoadVector256(pMatTemp += ccol));
-                            Vector256<float> vector = Avx.And(mask, Avx.LoadVector256(pSrcCurrent));
+                        Vector256<float> x01 = Avx.LoadAlignedVector256(pMatTemp);
```
Comment (tannergooding) [inline src/Microsoft.ML.CpuMath/AvxIntrinsics.cs]: nit: Codegen will be better on modern computers if we use `LoadVector256` (which does an unaligned load and allows it to be folded into the consuming instruction).

Comment (tannergooding) [reply src/Microsoft.ML.CpuMath/AvxIntrinsics.cs]: NOTE: Changing this to `LoadVector256` will also no longer throw if the input is unaligned, so adding a `Debug.Assert` or a helper might be beneficial. (I have a proposal on CoreFX to add a built-in helper that does this).

File: src/Microsoft.ML.CpuMath/AvxIntrinsics.cs [hunk 4]
```diff
@@ -450,326 +272,122 @@ Vector256<float> SparseMultiplicationAcrossRow()
...
-                            Vector256<float> leadingMask = Avx.LoadVector256(((float*)(pLeadingAlignmentMask)) + ((8 - remainder) * 8));
-                            Vector256<float> x3 = Avx.LoadVector256(pDstCurrent);
-                            x02 = Avx.Or(x02, Avx.And(x3, leadingMask));
-
-                            Avx.Store(pDstCurrent, x02);
-                            pDstCurrent += 8;
-                            pMatCurrent += 8;
-                        }
-                    }
+                    x02 = Avx.Multiply(x01, x02);
```
Comment (tannergooding) [inline src/Microsoft.ML.CpuMath/AvxIntrinsics.cs]: This one can use the `MultiplyAdd` helper, right?

File: test/Microsoft.ML.CpuMath.PerformanceTests/PerformanceTests.cs [hunk 5]
```diff
@@ -15,12 +15,18 @@ public abstract class PerformanceTests
         private const int ExponentMin = 0;
         private const int ExponentRange = ExponentMax / 8;
 
-        protected const int IndexLength = 1000003;
-        protected const int Length = 1000003;
-        protected const int MatrixIndexLength = 100;
+        protected const int IndexLength = 1000000;
```
Comment (tannergooding) [inline test/Microsoft.ML.CpuMath.PerformanceTests/PerformanceTests.cs]: nit: Might it be good to ensure this continues working even with inputs that have odd lengths?

File: test/Microsoft.ML.TimeSeries.Tests/TimeSeriesEstimatorTests.cs (hunk 1 above)
Comment (Anipik) [reply test/Microsoft.ML.TimeSeries.Tests/TimeSeriesEstimatorTests.cs]: The alignment is hardcoded to 16 bit in timeseries code which causes the debug.Assert statements to fail for avx. In avx, these statements check if the data is 32 bit aligned.
I will correct it in the next commit

File: src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs [hunk 6]
```diff
@@ -5,70 +5,425 @@
...
+using System.Runtime.InteropServices;
 
 namespace Microsoft.ML.Runtime.Internal.CpuMath
 {
     [BestFriend]
     internal static partial class CpuMathUtils
     {
         // The count of bytes in Vector128<T>, corresponding to _cbAlign in AlignedArray
-        private const int Vector128Alignment = 16;
+        private const int CbAlign = 16;
```
Comment (eerhardt) [inline src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs]: I think you should keep this name the same.

File: src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs [hunk 7]
```diff
@@ -5,70 +5,425 @@
...
 
         [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
         public static int GetVectorAlignment()
-            => Vector128Alignment;
+            => CbAlign;
 
-        public static void MatrixTimesSource(bool transpose, AlignedArray matrix, AlignedArray source, AlignedArray destination, int stride) => SseUtils.MatTimesSrc(transpose, matrix, source, destination, stride);
+        private static bool Compat(AlignedArray a)
+        {
+            Contracts.AssertValue(a);
```
Comment (eerhardt) [inline src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs]: Moving code like this is really hard to review.  Did you make any changes to this code, or is it a straight copy from Sse.cs to this file?

Comment (Anipik) [reply src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs]: it is straight copy from sse.cs

Comment (Anipik) [reply src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs]: The only addition is the function GetVectorAlignment

**Comment by Anipik** [comment]
@eerhardt can we merge this one ?

**Comment by eerhardt** [comment]
Since we are editing Time Series code, I think it would be good for @codemzs to sign off on the changes to it.

**Pull request merged**